### ISSUE TITLE
enzyme-adapter-react-16 definitions cannot be imported in to React TSX files

### ISSUE THREAD

**Issue by jimmyheaddon** [DefinitelyTyped/DefinitelyTyped#29655]
Examples of others having this issue can be found here - https://github.com/Microsoft/TypeScript-React-Starter/issues/76.  Essentially, because there are no named or default exports, within TSX files the imports don't seem to work as the examples suggest, but plain TS files seem happy enough.

For me, just changing `export = ReactSixteenAdapter;` to `export default ReactSixteenAdapter;` fixed the issue.

- [x] I tried using the `@types/enzyme-adapter-react-16` package (version 1.0.3) and had problems.
- [x] I tried using the latest stable version of tsc.
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  Perhaps not inappropriate, but others have the same issue, and came to the same conclusion.
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @tkrotoff

**Comment by tkrotoff** [comment]
Hi @jimmyheaddon I suggest you to create a PR and see if there is any regression with DefinitelyTyped unit tests + esModuleInterop

**Comment by jimmyheaddon** [comment]
Thanks @tkrotoff, I've created a PR of what I had in mind, but I'm obviously open to suggestions!  I've tested this on a handful of public and private repositories, and it's given me the additional functionality I was after, without breaking existing references.

NOTE: The use of allowSyntheticDefaultImports: true is required in tsconfig.json for ECMAScript syntax. Whilst this is not ideal, it does allow compatibility that would otherwise not be possible.
